test/framework/containers.py

```diff
@@ -282,6 +292,107 @@ def test_end2end_singularity_image(self):
         stdout, stderr = self.run_main(args)
         self.assertFalse(stderr)
         regexs[-3] = "^== Running 'sudo\s*SINGULARITY_TMPDIR=%s \S*/singularity build .*" % self.test_prefix
+
+    def test_end2end_dockerfile(self):
+        test_ecs = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'easyconfigs', 'test_ecs')
+        toy_ec = os.path.join(test_ecs, 't', 'toy', 'toy-0.0.eb')
+
+        containerpath = os.path.join(self.test_prefix, 'containers')
+        os.environ['EASYBUILD_CONTAINERPATH'] = containerpath
+        # --containerpath must be an existing directory (this is done to avoid misconfiguration)
+        mkdir(containerpath)
+
+        base_args = [
+            toy_ec,
+            '--containerize',
+            '--container-type=docker',
+            '--experimental',
+        ]
+
+        error_pattern = "Unsupported container base image 'not-supported'"
+        self.assertErrorRegex(EasyBuildError,
+                              error_pattern,
+                              self.eb_main,
+                              base_args + ['--container-base=not-supported'],
+                              raise_error=True)
+
+        for cont_base in ['ubuntu:16.04', 'centos:7']:
+            stdout, stderr = self.run_main(base_args + ['--container-base=%s' % cont_base])
+            self.assertFalse(stderr)
+            regexs = ["^== Dockerfile definition file created at %s/containers/Dockerfile.toy-0.0" % self.test_prefix]
+            self.check_regexs(regexs, stdout)
+            remove_file(os.path.join(self.test_prefix, 'containers', 'Dockerfile.toy-0.0'))
+
+        self.run_main(base_args + ['--container-base=centos:7'], raise_error=True)
+
+        error_pattern = "Container recipe at %s/containers/Dockerfile.toy-0.0 already exists, " \
+                        "not overwriting it without --force" % self.test_prefix
+        self.assertErrorRegex(EasyBuildError,
+                              error_pattern,
+                              self.eb_main,
+                              base_args + ['--container-base=centos:7'],
+                              raise_error=True)
+
+        remove_file(os.path.join(self.test_prefix, 'containers', 'Dockerfile.toy-0.0'))
+
+        base_args.insert(1, os.path.join(test_ecs, 'g', 'GCC', 'GCC-4.9.2.eb'))
+        self.run_main(base_args + ['--container-base=ubuntu:16.04'], raise_error=True)
+        def_file = read_file(os.path.join(self.test_prefix, 'containers', 'Dockerfile.toy-0.0'))
+        regexs = [
+            "FROM ubuntu:16.04",
+            "eb toy-0.0.eb GCC-4.9.2.eb",
+            "module load toy/0.0 GCC/4.9.2",
+        ]
+        self.check_regexs(regexs, def_file)
+        remove_file(os.path.join(self.test_prefix, 'containers', 'Dockerfile.toy-0.0'))
```

Comment:
not needed strictly speaking, the test-specific temporary directory at `self.test_prefix` is automatically cleaned up for each test